enterprise/backend/src/metabase_enterprise/transforms/util.clj

```diff
@@ -466,3 +466,25 @@
     (if tag-ids
       (filter #(some tag-ids (get-in % field-path)))
       identity)))
+
+(defn ->type-filter-xf
+  "Returns an xform for a transform source type filter."
+  [types]
+  (let [types (->> types (map keyword) set not-empty)]
+    (if types
+      (filter #(types (-> % :source :type keyword)))
```

Comment:
nit: can thread all the way, baby

probably clearer if you rename `types` to `keep-type?` or something.